editor/editor_help_search.cpp

```diff
@@ -590,27 +657,54 @@ TreeItem *EditorHelpSearch::Runner::_create_class_hierarchy(const ClassMatch &p_
 	return class_item;
 }
 
+bool EditorHelpSearch::Runner::_find_or_create_item(TreeItem *p_parent, const String &p_item_meta, TreeItem *&r_item) {
+	// Attempt to find in cache.
+	if (tree_cache->item_cache.has(p_item_meta)) {
+		r_item = tree_cache->item_cache[p_item_meta];
+
+		// Remove from cache.
+		tree_cache->item_cache.erase(p_item_meta);
```

Comment:
I wasn't sure why we would clear the cache here, since the HashMap should update existing items anyway. But removing this line doesn't introduce any performance improvements (and I may have introduced a leak with this, didn't debug much).

Reply:
It would risk double deleting items on close as any item already in the tree is deleted when clearing it, but might work out as it's deferred I realize, but unsure

Reply:
Well, doesn't matter if removing it doesn't help. Just thought I'd mention it.

Reply:
Thank you 🙂